automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/WorkspaceDataPage.scala

```diff
@@ -50,22 +52,36 @@ class WorkspaceDataPage(namespace: String, name: String)(implicit webDriver: Web
     * @param downloadPath the directory where the browser saves downloaded files
     * @return the relative path to the moved download file, or None if downloadPath was not given
     */
-  def downloadMetadata(downloadPath: Option[String] = None): Option[String] = {
+  def downloadMetadata(downloadPath: Option[String] = None): Option[String] = synchronized {
 
     def archiveDownloadedFile(sourcePath: String): String = {
+      // wait up to 5 seconds for file exist
+      val f = new File(sourcePath)
+      logger.info(s"Retrying - downloaded file wait for exist: $f")
+      retry[Boolean](1.seconds, 10.seconds) {
+        if (f.exists()) {
+          Some(true)
+        } else {
+          None
+        }
+      }
       val date = new SimpleDateFormat("yyyy-MM-dd_HH-mm-ss-SSS").format(new java.util.Date())
       val destFile = new File(sourcePath).getName + s".$date"
       val destPath = s"downloads/$destFile"
       Util.moveFile(sourcePath, destPath)
+      logger.info(s"Moved file. sourcePath: $sourcePath, destPath: $destPath")
       destPath
     }
 
-    downloadMetadataButton.doClick()
+    downloadMetadataButton.awaitVisible()
+
     /*
      * Downloading a file will open another window while the download is in progress and
      * automatically close it when the download is complete.
      */
-    await condition (windowHandles.size == 1, 60)
+    // await condition (windowHandles.size == 1, 30)
+    // .submit call takes care waiting for a new window
+    find(tagName("form")).get.underlying.submit()
```

Comment:
The form is in data table `entity-table`. I can restrict the scope of form by css. What do you think of this?
`val form = CssSelectorQuery(s"{dataTable.query.queryString} form")`
`find(form).get.underlying.submit()`

Reply:
Yes, that definitely helps. Don't we have some functions for finding things inside other things? If not, what you proposed is fine. It'll help keep this test from failing when someone in the future adds a form to the page (this test failure would be a distraction because it would be completely unrelated to the change).